pylint/config/config_initialization.py

```diff
@@ -83,7 +83,17 @@ def _config_initialization(
         msg = ", ".join(unrecognized_options)
         linter._arg_parser.error(f"Unrecognized option found: {msg}")
 
-    # Set the current module to configuration as we don't know where
+    # Now that config file and command line options have been loaded
+    # with all disables, it is safe to emit messages
+    linter.set_current_module(str(config_file) if config_file else None)
```

Comment:
We have deprecated passing `None` to `set_current_module`. So it should probably be `""`.